05-Jason-Luo/wk04/2-tue/mta.rb

```diff
@@ -0,0 +1,97 @@
+# "Change at Union Square."
+# "Your journey continues through the following stops: 23rd, 28th, 33rd."
+# "7 stops in total."
+
+trainlines = {
+    :N => ["Times Square", "34th", "N-28th", "N-23rd", "Union Square", "N-8th"],
+    :L => ["L-8th", "6th", "Union Square", "3rd", "1st"],
+    :S => ["Grand Central", "33rd", "6-28th", "6-23rd", "Union Square", "Astor Place"]
+}
+
+def stationToUnion(SelectLine, SelectUnion, SelectStation)
+    if SelectStation > SelectUnion 
+        return "#{(SelectLine[SelectUnion..SelectStation]).reverse()}\n\n" # when SelectStation > SelectUnion, slice from SelectUnion, then reverse the array to shuffle back to correct order. 
+        # Add 1 when slicing as it is off by 1
+    else 
+        BACKTRIP = "#{(SelectLine[SelectStation...SelectUnion]).reverse()}\n\n" #find what stations are in between the two stations on the return trip
+        # Reverse array to get correct order, then splice off 'Union Square' from array, as it is unneccessary
+        BACKTRIP.slice!(BACKTRIP[SelectUnion], 1)
+        return BACKTRIP    
+end
+
+def firstTrip(Line, Station) 
+    SelectLine = trainlines[Line]
+    SelectStation = trainlines[Line].index(Station)
+    SelectUnion = trainlines[Line].index("Union Square") #isolate Union square for QoL
+
+    if (SelectStation > SelectUnion) 
+        return stationToUnion(SelectLine, SelectUnion, SelectStation)
+    else 
+        return SelectLine[SelectStation..SelectUnion] #find what stations are in between the two stations
+        # Union Square is included in the first trip, to record total stops.
+end
+
+def secondTrip(Line, Station) 
+    SelectLine = trainlines[Line]
+    SelectStation = trainlines[Line].index(Station)
+    SelectUnion = trainlines[Line].index("Union Square") #isolate Union square for QoL
+
+    if (SelectStation < SelectUnion) 
+        return stationToUnion(SelectLine, SelectUnion, SelectStation)
+    else 
+        BACKTRIP = SelectLine[SelectUnion..SelectStation] #find what stations are in between the two stations on the return trip
+        # Slice from SelectUnion to SelectStation plus 1, to account for being one off
+        BACKTRIP.slice!(BACKTRIP[SelectUnion], 1) # Splice off Union Square as it is not necessary for second trip log
+        return BACKTRIP
+end
+
+def sameLine(Line, StationStart, StationEnd) 
+    SelectLine = trainlines[Line]
+    TopStation = trainlines[Line].index(StationStart)
+    BottomStation = trainlines[Line].index(StationEnd) #isolate Union square for QoL
+
+    if (BottomStation < TopStation) 
+        return "#{(SelectLine[BottomStation..TopStation]).reverse()}\n\n" 
+        #slice from end (BottomStation), then reverse Array to get correct order of Array
+    else 
+        return SelectLine[TopStation..BottomStation]
+        #if BottomStation > TopStation, return array plus the last stop
+end
+
+def startEndUnion(LineStart, StationStart, LineEnd, StationEnd) 
+    if (StationStart === "Union Square") 

```

Comment:
```suggestion
    if (StationStart == "Union Square") 
```
You don't need to use strict equality (`===`) in Ruby like you do in Javascript.